### DISCUSSION TITLE
Define CHECK constraints at the Model level

### DISCUSSION THREAD

**Discussion by LebonNic** [sequelize/sequelize#16750] [Help & Questions] [answered]
Hello,

I would like to know if there is a way of defining a CHECK constraint at Model level.

Currently, the only solution I know of is to add this type of constraint via the `QueryInterface` in a migration file. Here's an example:

```javascript
module.exports = {
  async up(queryInterface, Sequelize) {
    await queryInterface.addConstraint("product", ["price"], {
      type: "check",
      name: "price_check",
      fields: ["price"],
      where: {
        price: {
          [Sequelize.Op.gt]: 0,
        },
      },
    });
  },

  async down(queryInterface, Sequelize) {
    await queryInterface.removeConstraint("product", "price_check");
  },
};
```

What I want to do here is define the constraint at the Model level so that it is applied at the time of `sequelize.sync({ force: true });
`
I know there's a validator option, but that's not what I'm looking for here. I really want the constraint to be checked at the database engine level and not at the ORM level.

Nicolas

**Comment by ephys** [comment, marked as answer]
That is not currently supported: https://sequelize.org/docs/v7/models/validations-and-constraints/#check-constraints
